article/article.md

```diff
@@ -0,0 +1,153 @@
+
+Over time we grew dissatisfied with this approach, however. Not only was Carto challenging to set up and maintain, particularly in our [containerized microservice implementation](https://github.com/ihmeuw/cartodb-docker), but the tile server was quite slow at producing the colorized map tiles, making using the visualization tool a frustrating experience. We did some performance testing of the server with [Apache Jmeter](https://jmeter.apache.org/) and found that it was taking about 3 seconds, in the best case, to return a tile. I suspect we could have squeezed better performance out of the tile server with some optimizations, but its codebase was always something of an impenetrable black box to us, and the prospect of searching for needles in this haystack seemed quite daunting.
+
+## The Idea: Client-side rendering with WebGL
+
+Around this time, I began thinking about other ways to render our map tiles. One option was to create our own tile server to do the colorization. A custom server could be better suited to our specific needs, and, crucially, we'd be more able optimize its performance because we'd understand it better. Yet a server-side solution had some inherent limitations. Rendering a large number of pixels on the CPU is bound to be slow, because the pixels must be processed sequentially (or with limited parallelization taking advantage of multicore architecture). With commodity hardware and the VMs in which our containerized services run, there would be hard limits on how fast the tiles could be produced. It seemed to me that the ideal way to render the images was on the [GPU](https://en.wikipedia.org/wiki/Graphics_processing_unit), which is specifically designed to process large numbers of pixels in parallel. Fortunately, modern browsers provide a way to do GPU processing: [WebGL](https://developer.mozilla.org/en-US/docs/Web/API/WebGL_API)! A WebGL solution would mean the tile server would have very little to do - just retrieve pixel data from PostGIS and then pass it in essentially raw form to the client. The user's browser would then be responsible for actually colorizing the pixels, and it could take advantage of GPU rendering to do this very efficiently.
+
+Having the raw floating-point pixel values available clientside would have another advantage as well - allowing the user quick access to the actual pixel-level estimates in the visualization. While humans generally have a pretty keen color sense, the result of mentally translating a color from a gradient back to the value it's supposed to represent seemed pretty imprecise. Our visualization already had the capability of getting the data value of a given pixel when clicked by the user, but doing so required a backend request and was therefore relatively slow. Having this data available clientside would mean we could retrieve and display these values very quickly.
```

Comment:
"clientside" should probably be two words.

This sentence is fairly obvious, distracting, and doesn't add much to what's interesting about the technical solution you came up with: "While humans generally have a pretty keen color sense, the result of mentally translating a color from a gradient back to the value it's supposed to represent seemed pretty imprecise." I would just say that the users need to see the precise values that the pixels represented in a tooltip, and then explain why your solution is an improvement over the backend request.

Reply:
- hyphenated "client-side"
- removed the sentence to which you objected
- revised the last sentence in the paragraph to clarify that the approach would enable showing the value in a tooltip on hover